src/role-resolver.js

```diff
@@ -0,0 +1,70 @@
+angular.module('mmRoute')
+.factory('mmRoleResolver', function ( $window, $injector, mmRoute ) {
+
+  function checkTemplatePermission( accessRole ) {
+    var roles = $injector.invoke(mmRoute.roleGetter, $window);
+
+    if ( !Array.isArray(accessRole) ) {
+      return roles.indexOf(accessRole) !== -1;
+    } else {
+      return accessRole.every(function ( singleRole ) {
+        return roles.indexOf(singleRole) !== -1;
+      });
+    }
+  }
+
+  function chooseTemplate( accessObjects ) {
+    var chosenTemplate = null;
+
+    for ( var i = 0; i < accessObjects.length; i++ ) {
+      var accessObject = accessObjects[ i ];
+      if ( accessObject.roles.some(checkTemplatePermission) ) {
+        chosenTemplate = accessObject.page;
+        break;
+      }
+    }
+
+    if ( !chosenTemplate ) {
+      throw new Error();
+    }
+
+    return chosenTemplate;
+
+  }
+
+  return {
+    chooseTemplate: chooseTemplate,
+    _checkTemplatePermission: checkTemplatePermission
```

Comment:
Because it is only used from within the factory and I am only returning it so that I can test it.